lib/qa/authorities/discogs/discogs_instance_builder.rb

```diff
@@ -0,0 +1,127 @@
+          end
+        end
+        stmts # w/out this line, building the graph throws an undefined method `graph_name=' error
+      end
+
+      def build_format_characteristics(df, count)
+        stmts = []
+        case df["type"]
+        when "playbackChannel"
+          stmts << contruct_stmt_uri_object("Instance1", "http://id.loc.gov/ontologies/bibframe/soundCharacteristic", df["uri"])
+          stmts << contruct_stmt_uri_object(df["uri"], rdf_type_predicate, "http://id.loc.gov/ontologies/bibframe/PlaybackChannel")
+          stmts << contruct_stmt_literal_object(df["uri"], rdfs_label_predicate, df["label"])
+        when "dimension"
+          stmts << contruct_stmt_uri_object("Instance1", "http://id.loc.gov/ontologies/bibframe/dimensions", df["label"])
+        when "playingSpeed"
+          stmts << contruct_stmt_uri_object("Instance1", "http://id.loc.gov/ontologies/bibframe/soundCharacteristic", "PlayingSpeed#{count}")
+          stmts << contruct_stmt_uri_object("PlayingSpeed#{count}", rdf_type_predicate, "http://id.loc.gov/ontologies/bibframe/PlayingSpeed")
+          stmts << contruct_stmt_literal_object("PlayingSpeed#{count}", rdfs_label_predicate, df["label"])
+        end
+        stmts # w/out this line, building the graph throws an undefined method `graph_name=' error
+      end
+
+      def build_format_name_stmts(name)
+        stmts = []
+        # map discogs name field to the corresponding LOC carrier
+        return stmts unless name.present?
+        dc = discogs_formats[name.gsub(/\s+/, "")]
+        if dc.present?
+          stmts << contruct_stmt_uri_object("Instance1", "http://id.loc.gov/ontologies/bibframe/carrier", dc["uri"])
+          stmts << contruct_stmt_uri_object(dc["uri"], rdf_type_predicate, "http://id.loc.gov/ontologies/bibframe/Carrier")
+          stmts << contruct_stmt_literal_object(dc["uri"], rdfs_label_predicate, dc["label"])
+          stmts.concat(build_base_materials(name))
+        else
+          # if it's not a carrier, it's an edition statement
+          stmts << contruct_stmt_literal_object("Instance1", "http://id.loc.gov/ontologies/bibframe/editionStatement", name)
+        end
+        stmts # w/out this line, building the graph throws an undefined method `graph_name=' error
+      end
+
+      def build_base_materials(name)
+        stmts = []
+        return stmts unless name == "Vinyl" || name == "Shellac"
+        id = name == "Vinyl" ? "300014502" : "300014918"
+
+        stmts << contruct_stmt_uri_object("Instance1", "http://id.loc.gov/ontologies/bibframe/baseMaterial", "http://vocab.getty.edu/aat/" + id)
+        stmts << contruct_stmt_uri_object("http://vocab.getty.edu/aat/" + id, rdf_type_predicate, "http://id.loc.gov/ontologies/bibframe/BaseMaterial")
+        stmts << contruct_stmt_literal_object("http://vocab.getty.edu/aat/" + id, rdfs_label_predicate, name)
+        stmts # w/out this line, building the graph throws an undefined method `graph_name=' error
```

Comment:
This one I'm surprised by.  The `<<` operator on an array returns the updated array.  But its fine to have stmts here.  It makes it clear that stmts is being returned.

Reply:
Removed the closing line.